doc/Clock commands/clock_managment_hld.md

```diff
@@ -0,0 +1,180 @@
+1.	Set/show date-time command
+2.	Set/show timezone command
+
+
+### 1.3 <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations 
+
+N/A
+
+### 1.4 Overview 
+
+The clock commands allow to set and review the current time parameters of the system - including: time, date and timezone. 
+
+### 1.5 Requirements
+####  1.5.1. <a name='FunctionalRequirements'></a>Functional requirements
+
+1. Any time configuration that will change in the system will change the system time. 
+
+####  1.5.2. <a name='ConfigurationManagementRequirements'></a>Configuration and Management Requirements
+The requirements from the module are: 
+1.	Set and show the system time and date.
+2.	Set and show the system timezone.
+
+
+##  2 <a name='Design'></a>Design
+
+###  2.1 <a name='High-LevelDesign'></a>High-Level Design
+
+The design of this feature is based on Linux command of <b>timedatectl</b>.<BR>
+(man page for timedatectl: https://man7.org/linux/man-pages/man1/timedatectl.1.html)
+
+All set operations are based on this command, executing directly in linux upon user CLI execution.
+The time/date/timezone are configured directly to Linux, and is persistent upon any reboot.
+
+System state of time/date/timezone will appear in 1 line as an output of existing "show clock" command.
+
+Set operations will be divided into 2 different commands:
+
+1. config clock set-timezone <timezone> (to set timezone command)
+	* will get single input as a string, and validate it is valid as part of ("timedatectl list-timezones" command).
+	Validation will be done either by YANG model, or (if not possible) by issueing relevant error to log.
+	* Value will be stored in db for future upgrade operations.
+	  Value is persistent upcon reboot.
+	* Linux timedatectl with set-timezone flag will be called.
+	  e.g. timedatectl set-timezone "Asia/Kolkata"
+	* <b>In case NTP is enabled -> timezone configuration is allowed and overrides the current time.</b>
+
+
+2. config clock set-date "<YYYY-MM-DD HH:MM:SS>" (to set time and date)
+	* Command will recieve single string with date-time format "<YYYY-MM-DD HH:MM:SS>"
+	* It will be possible to call command with the following options:
+		1.	Only with a date <YYYY-MM-DD>
+		2. 	Only with time <HH:MM:SS>
+		3.  both date and time "<YYYY-MM-DD HH:MM:SS>"
+	* Command does not need to be stored in DB.
+	* Linux timedatectl with set-time flag will be called.
+	  e.g. timedatectl set-time "2012-10-30 18:17:16"
+	* <b>In case NTP is enabled -> time/date set is NOT allowed and being blocked</b>
+
+
+Both set commands will be written directly to Linux (via imedatectl command), and will be activated immediately.
+
+
+
+##  3 <a name='Configurationandmanagement'></a>Configuration and management
+
+###  3.1 <a name='ConfigDBTables'></a>ConfigDB Tables
+
+Only timezone configuration will be saved.
+Additional field will be added to DEVICE_METADATA table
+
+```
+DEVICE_METADATA :{
+    "timezone": {{string}}
+}
```

Comment:
We should reboot the rsyslog service inside the container in order to reflect the time change, please test and update here.

Reply:
If the future, if there is support for additional attributes to the clock besides timezone,  is it better to put timezone in a separate table other than inside DEVICE_METADATA? For example, if timezone is allowed to learn from DHCP, and a flag can indicate if the timezone from DHCP is preferred over statically configured timezone.

Reply:
> We should reboot the rsyslog service inside the container in order to reflect the time change, please test and update here.

Yes, rsyslog will be restarted. Added a relevant comment in HLD bottom of section 2.1.
Also added relevant case in Tests section below.

Reply:
> If the future, if there is support for additional attributes to the clock besides timezone, is it better to put timezone in a separate table other than inside DEVICE_METADATA? For example, if timezone is allowed to be learned from DHCP, and a flag can be used to indicate if the timezone from DHCP is preferred over statically configured timezone.

Currently, there is no such plan, and in case wanted to get from network the time, it could be done via NTP.
If in future it would be decided by community to extend, could also be adding a field to current table, or think in future if should be extended.
For now, it seems like a clear approach to put as part of METADATA table.

Reply:
Could you please elaborate how it can be done via NTP, if the timezone comes from dhcp?

Reply:
NTP will provide the time from network to be joint for all systems using it (instead of getting the time configuration from DHCP) - thus, you could get the same result from a single source configuring the time to all systems using it.